README.md

```diff
@@ -46,16 +46,42 @@ The config passed to Configure can contain the following fields.
 | `urls`                     | A list of connection URLs joined by comma. Must be a valid URLs.<br />Examples:<br />`nats://127.0.0.1:1222`<br />`nats://127.0.0.1:1222,nats://127.0.0.1:1223`<br />`nats://myname:password@127.0.0.1:4222`<br />`nats://mytoken@127.0.0.1:4222`                                                                                                                                                                                                                                                                                                                                                                | **true** |                                    |
 | `streamName`               | A stream name. It can contains alphanumeric characters and max length is 32 characters.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          | **true** |                                    |
-| `subject`                  | A name of a subject from which or to which the connector should read write.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      | **true** |                                    |
+| `subject`                  | A name of a subject from which or to which the connector should read write. It's possible to specify a subject that is different from the one belonging to the stream you specified, the connector in this case will handle messages properly.                                                                                                                                                                                                                                                                                                                                                                   | **true** |                                    |
```

Comment:
Sorry, I guess I misrepresented this behavior. I meant that it is possible to specify a name of a subject that belongs to a stream, but not to the stream you specified, and the connector will still read messages from the subject. This is the way how the nats.go library works by default.

Another option here is to retrieve `StreamInfo` by a stream name and then check if its `subjects` slice contains the subject you specified in the config and if it is not - raise an error.

What do you think? Should I add the check?

Reply:
I've removed the `stream` parameter and added a new one, `deliverSubject`, which is `<durable>.conduit` by default. The `stream` field was only used in constructing the deliver subject during the channel subscription, so yeah, having the `deliverSubject` instead of the `stream` makes much more sense.